### PR TITLE
fix several Sentry issues

### PR DESCRIPTION
**Pull request by Apollon77**
IOBROKER-JS-CONTROLLER-215, IOBROKER-JS-CONTROLLER-27G, IOBROKER-JS-CONTROLLER-24K, IOBROKER-JS-CONTROLLER-279, IOBROKER-JS-CONTROLLER-270, IOBROKER-JS-CONTROLLER-263, IOBROKER-JS-CONTROLLER-251, IOBROKER-JS-CONTROLLER-26M, IOBROKER-JS-CONTROLLER-25Y, IOBROKER-JS-CONTROLLER-253, IOBROKER-JS-CONTROLLER-24R, IOBROKER-JS-CONTROLLER-252, and some more

### CODE REVIEW

File: packages/controller/main.js [hunk 1]
```diff
@@ -900,27 +903,43 @@ function reportStatus() {
...
-        from: id
-    });
-    states.setState(id + '.memHeapUsed', {
-        val: Math.round(mem.heapUsed / 10485.76 /* 1MB / 100 */) / 100,
-        ack: true,
-        from: id
-    });
+    try {
+        const mem = process.memoryUsage();
+        states.setState(`${id}.memRss`, {
```
Comment (foxriver76) [inline packages/controller/main.js:925]: Has setState thrown then it will still throw async without await

File: packages/controller/main.js [hunk 2]
```diff
@@ -900,27 +903,43 @@ function reportStatus() {
...
+            ack: true,
+            from: id
+        });
+        states.setState(id + '.memHeapUsed', {
+            val: Math.round(mem.heapUsed / 10485.76 /* 1MB / 100 */) / 100,
+            ack: true,
+            from: id
+        });
+    } catch (e) {
+        logger.error(`${hostLogPrefix} Cannot read memoryUsage data : ${e.message}`);
```
Comment (foxriver76) [inline packages/controller/main.js:941]: Ugly blank ;-)

File: packages/controller/main.js [hunk 3]
```diff
@@ -2230,31 +2249,36 @@ async function processMessage(msg) {
...
+                        logger.info(`${hostLogPrefix} ${tools.appName} exit ${exitCode}`);
+                        if (msg.from) {
+                            sendTo(msg.from, 'cmdExit', { id: msg.message.id, data: exitCode });
+                            // Sometimes finished command is lost, recent it
+                            setTimeout(() => sendTo(msg.from, 'cmdExit', { id: msg.message.id, data: exitCode }), 1000);
+                        }
                     });
+                } catch (e) {
+                    logger.error(`${hostLogPrefix} ${tools.appName} ${e.message}`);
+                    msg.from && sendTo(msg.from, 'cmdStderr', { id: msg.message.id, data: e.message });
```
Comment (foxriver76) [inline packages/controller/main.js:2280]: ^^

File: packages/controller/main.js (hunk 1 above)
Comment (Apollon77) [reply packages/controller/main.js:925]: No the "process.memoryUsage" has thrown

File: packages/controller/main.js (hunk 3 above)
Comment (Apollon77) [reply packages/controller/main.js:2280]: ??

File: packages/controller/main.js (hunk 2 above)
Comment (Apollon77) [reply packages/controller/main.js:941]:
```suggestion
        logger.error(`${hostLogPrefix} Cannot read memoryUsage data: ${e.message}`);
```

Comment (Apollon77) [reply packages/controller/main.js:941]: Mean this?

Comment (foxriver76) [reply packages/controller/main.js:941]: Yes

File: packages/controller/main.js (hunk 3 above)
Comment (foxriver76) [reply packages/controller/main.js:2280]: Nit: true && fun()

i know it is on the legacy code above too but we could change if we add new lines ;-)

Comment (Apollon77) [reply packages/controller/main.js:2280]: ok, next time

**Pull request merged**